jetson/nav/search/searchStateMachine.cpp

```diff
@@ -53,14 +56,51 @@ NavState SearchStateMachine::executeSearch() {
     } else {
         // Either target works
         bool isWantedTarget = (lastWaypoint.id == leftTarget.id && leftTarget.id >= 0) ||
-                                (lastWaypoint.id == rightTarget.id && rightTarget.id >= 0);
+                              (lastWaypoint.id == rightTarget.id && rightTarget.id >= 0);
         if (isWantedTarget) {
             return NavState::DriveToTarget;
         }
     }
 
-    Odometry const& nextSearchPoint = mSearchPoints.front();
     double dt = sm->getDtSeconds();
+
+    // Recovery watch for search
+    if (sm->getRecovery()->getRecoveryState()) {
```

Comment:
Instead of typing `sm->getRecovery()` a billion times consider making a variable. Same idea with path and what not. This function is a little hard to read.